packages/lesswrong/server/scripts/resolveReviewMarkets.ts

```diff
@@ -0,0 +1,90 @@
+import { Posts } from "@/lib/collections/posts";
+import { manifoldAPIKeySetting } from "@/lib/instanceSettings";
+import { postGetMarketInfoFromManifold, LiteMarket } from "@/lib/collections/posts/annualReviewMarkets";
+import { Globals } from "@/lib/vulcan-lib";
+
+const manifoldAPIKey = manifoldAPIKeySetting.get()
+
+export const getMarketInfoFromManifold = async (marketId: string): Promise<LiteMarket | null > => {
+
+  const result = await fetch(`https://api.manifold.markets./v0/market/${marketId}`, {
+    method: "GET",
+    headers: {
+      "content-type": "application/json"
+    },
+  })
+  
+  if (!result.ok) {
+    //eslint-disable-next-line no-console
+    console.error(`HTTP error! status: ${result.status}`); 
+    return null
+  }
+
+  return result.json()
+}
+
+const resolvePosts = async (year: number, limit?: number) => {
+
+  const resolutionLimitOptions = limit ? {limit} : {}
+
+  const postsToResolve = await Posts.find({
+    manifoldReviewMarketId: {$exists: true},
+    createdAt: {$gte: new Date(`${year}-01-01`), $lte: new Date(`${year}-12-31`)}
```

Comment:
I think this wants to be `$lt: new Date(\`${year + 1}-01-01\`)`; else any post made after 00:00 UTC(?) on 12-31 will be excluded

Reply:
Really? Even though it's lte no lt? is this a timezone thing?

Reply:
I mean I could be wrong here, need to test it in a node repl

Reply:
Yeah I think this is real:
```
> foo = new Date('2023-12-31')
2023-12-31T00:00:00.000Z
> foo.getHours()
16
> foo.getUTCHours()
0
> foo.getTime()
1703980800000
> foo.setUTCHours(1)
1703984400000
> foo.setUTCHours(23)
1704063600000
```

Reply:
🫡

Reply:
It's not an implementation detail of collection date handling, it's just how Javascript works. There's no such thing as a "date" in Javascript without a time - when you omit a time is just sets it to 00:00 automatically. `new Date("2023-12-31T00:00:00.000Z").getTime() === new Date("2023-12-31").getTime()` is `true`.

Reply:
You misunderstand me, @oetherington. This date is going to be compared at some point to the date on the records being received. This could be via comparing two JavaScript Date objects using the built in `<`, or it could not be. For example, it could be (and I'd guess is?) translated to a SQL query of some sort, and that might or might not use the full datetime

Reply:
Sure, I get what you mean, but as soon as you do ``new Date(`${year}-12-31`)`` it will always be comparing against ``new Date(`${year}-12-31T00:00:00.000Z`)`` - dates without times don't exist. Postgres does have a `DATE` type but it's not used anywhere in the codebase - all dates are stored as `TIMESTAMPTZ`.

Reply:
Good to know about the `TIMESTAMPTZ`, that definitely answers the top-level question.

But I don't think the first part of your response makes sense as a response? Like, we agree about the equality you mentioned in your previous response. But surely you agree that I can construct a comparison function `dateLte : Date -> Date -> Boolean` that doesn't take into account the time part of a `Date`?

Reply:
Yeah, it's not incoherent to consider a "date-only" comparison, though I've never seen that pattern used anywhere

Reply:
Err, ok, that's probably false.  But I don't think I've ever seen it used this close to the storage layer, at least.

Reply:
Oh right, I think I see where you're coming from now. Yes, it'd be possible to contruct that function, but I think it'd be _extremely_ unusual for that to be the default behaviour for a function that has no way to know whether or not you specified a date (unusual enough that it didn't even occur to me that you could have meant that 😅).

Reply:
It's probably reasonable that I should have known that this field would include time info, but I don't actually think it would be unusual for a collection management library (this is calling a function on the generated `Posts` collection) to use a custom date comparator on date-typed fields